src/main/scala/com/typesafe/sbt/packager/docker/Keys.scala

```diff
@@ -57,4 +57,10 @@ private[packager] trait DockerKeysEx extends DockerKeys {
   lazy val dockerAdditionalPermissions =
     taskKey[Seq[(DockerChmodType, String)]]("Explicit chmod calls to some of the paths.")
   val dockerApiVersion = TaskKey[Option[DockerApiVersion]]("dockerApiVersion", "The docker server api version")
+  val dockerLayerGrouping = settingKey[String => Int](
+    "Group files by path into in layers to increase docker cache hits. " +
+      "Lower index means the file would be a part of an earlier layer."
+  )
+  val dockerLayerMappings =
+    taskKey[Seq[(Int, File, String)]]("List of layer, source file and destination in Docker image.")
```

Comment:
Move those to [`DockerKeysEx`](https://github.com/sbt/sbt-native-packager/blob/dec9481686ae559a9a72be6a7440bca6ce6ae90a/src/main/scala/com/typesafe/sbt/packager/docker/Keys.scala#L54), which solves the binary incompat issue.